storage/innobase/handler/ha_innodb.cc

```diff
@@ -2916,7 +2916,11 @@ ha_innobase::ha_innobase(
 		  ),
 	m_start_of_scan(),
         m_mysql_has_locked()
-{}
+{
+	if (table_arg && table_arg->row_type == ROW_TYPE_REDUNDANT) {
+		m_int_table_flags |= HA_EXTENDED_TYPES_CONVERSION;
```

Comment:
@dr-m said in Slack that sql layer may have incorrect `ROW_FORMAT` value and only InnoDB know it exactly. Thus this code is incorrect. You probably need to call `dict_table_is_comp()` somewhere here.

Reply:
Good point. Fixed.